homeassistant/helpers/json.py

```diff
@@ -188,10 +195,23 @@ def save_json(
         formatted_data = format_unserializable_data(
             find_paths_unserializable_data(data, dump=dump)
         )
-        msg = f"Failed to serialize to JSON: {filename}. Bad data at {formatted_data}"
+        destination = f": {destination}" if destination else ""
+        msg = f"Failed to serialize to JSON{destination}. Bad data at {formatted_data}"
         _LOGGER.error(msg)
```

Comment:
Maybe we should let the caller catch and log the error instead of doing that here. Also, the caller knows the destination file (if any), I guess we shouldn't pass it in?